### PR TITLE
[TACHYON-1430] Add the checkstyle rule FinalClass

### PR DESCRIPTION
**Pull request by luoli523**
[TACHYON-1430] Add the checkstyle rule FinalClass

JIRA link https://tachyon.atlassian.net/browse/TACHYON-1430

Add the FinalClass checkstyle module in build/checkstyle/tachyon_checks.xml to enable style checks that a class which has only private constructors is declared as final. And with this check, find several classes not fit the rule.

There are:
clients/unshaded/src/main/java/tachyon/client/file/TachyonFileSystem.java
clients/unshaded/src/main/java/tachyon/client/table/TachyonRawTables.java
clients/unshaded/src/main/java/tachyon/hadoop/ConfUtils.java
common/src/main/java/tachyon/exception/PreconditionMessage.java
integration-tests/src/test/java/tachyon/client/StreamOptionUtils.java
servers/src/main/java/tachyon/Format.java
servers/src/main/java/tachyon/ValidateConf.java
servers/src/main/java/tachyon/master/TachyonMaster.java
servers/src/main/java/tachyon/master/file/meta/options/CreatePathOptions.java
servers/src/main/java/tachyon/web/WebInterfaceGeneralServlet.java
servers/src/main/java/tachyon/web/WebInterfaceWorkersServlet.java
servers/src/main/java/tachyon/worker/block/BlockMetadataManager.java
servers/src/main/java/tachyon/worker/block/ClientRWLock.java

### CODE REVIEW

**Comment by jsimsa** [comment]
@luoli523 thank you for your PR, you will need to use PowerMock in BlockLockManagerTest to make it possible to mock a final class

**Comment by jsimsa** [comment]
just noticed your 2nd commit ... LGTM

**Comment by jsimsa** [comment]
By the way, your commits aren't linked to a valid Github account yet. For us to properly credit your contributions, please follow these instructions to fix the issue: https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/

**Comment by luoli523** [comment]
Ok, thank you @jsimsa

**Comment by jsimsa** [comment]
after you fix up the commits in your local repository, you will need to use "push -f" to update the commits in the PR

File: servers/src/test/java/tachyon/worker/block/BlockLockManagerTest.java [hunk 1]
```diff
@@ -41,8 +46,8 @@
 
   @Before
   public void before() throws Exception {
-    BlockMetadataManager mockMetaManager = Mockito.mock(BlockMetadataManager.class);
-    Mockito.when(mockMetaManager.hasBlockMeta(TEST_BLOCK_ID)).thenReturn(true);
+    BlockMetadataManager mockMetaManager = PowerMockito.mock(BlockMetadataManager.class);
+    PowerMockito.when(mockMetaManager.hasBlockMeta(TEST_BLOCK_ID)).thenReturn(true);
```
Comment (aaudiber) [inline servers/src/test/java/tachyon/worker/block/BlockLockManagerTest.java]: This can still be `Mockito.when`

**Comment by aaudiber** [comment]
@luoli523 Thanks for improving our style conventions! I left a minor comment, otherwise LGTM.

**Comment by yupeng9** [comment]
Thanks for adding this style-check rule! LGTM.

**Comment by jsimsa** [comment]
@luoli523 this PR is good to me merged when you link your commits to your GitHub account

**Comment by luoli523** [comment]
@jsimsa , I am already linked my GitHub account, thank you

**Comment by jsimsa** [comment]
@luoli523 the commits on this PR still have the wrong author, follow these instructions (http://stackoverflow.com/questions/750172/change-the-author-of-a-commit-in-git) to use git rebase to fix the author of those commits in your local repository and then use "git push -f" to update this PR

**Comment by jsimsa** [comment]
in particular, if you look here: https://github.com/amplab/tachyon/pull/2270/commits, you will see that only the last commit is linked to your account

**Comment by jsimsa** [comment]
This could also be helpful: https://help.github.com/articles/changing-author-info/

**Comment by luoli523** [comment]
Done, Thanks a lot for the help. @jsimsa

**Comment by jsimsa** [comment]
LGTM

**Pull request merged**